.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,15 @@
+name: Shell Lint
+
+on: [pull_request, push]
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install ShellCheck
+        run: sudo apt-get install -y shellcheck
+      - name: Run ShellCheck
+        run: |
+          find . -name '*.sh' -exec shellcheck {} +
```

Comment:
we should probably exclude some codes. In test definitions the following codes are excluded: `SC1091 SC2230 SC3043`